packages/web-app-external/src/App.vue

```diff
@@ -73,8 +83,15 @@ export default defineComponent({
       })
     }
 
-    const loadAppUrl = useTask(function* () {
+    const loadAppUrl = useTask(function* (signal, viewMode: string) {
       try {
+        if (props.isReadOnly && viewMode === 'write') {
```

Comment:
would it make sense to create an enum for the viewMode?

Reply:
seems overkill to me tbh. In theory we could also support more modes which are defined by the backend... this is just some specific handling for an edge case